client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -26,35 +26,82 @@ import './style.scss';
 
 export class SiteVerticalsSuggestionSearch extends Component {
 	static propTypes = {
+		autoFocus: PropTypes.bool,
 		charsToTriggerSearch: PropTypes.number,
-		initialValue: PropTypes.string,
+		defaultVertical: PropTypes.object,
 		onChange: PropTypes.func,
 		placeholder: PropTypes.string,
-		shouldShowPopularTopics: PropTypes.func,
 		searchValue: PropTypes.string,
+		shouldShowPopularTopics: PropTypes.func,
 		verticals: PropTypes.array,
-		defaultVertical: PropTypes.object,
 	};
 
 	static defaultProps = {
+		autoFocus: false,
 		charsToTriggerSearch: 1,
-		initialValue: '',
+		defaultVertical: {},
 		onChange: noop,
 		placeholder: '',
-		shouldShowPopularTopics: noop,
 		searchValue: '',
+		shouldShowPopularTopics: noop,
 		verticals: [],
-		defaultVertical: {},
 	};
 
 	constructor( props ) {
 		super( props );
 		this.state = {
 			railcar: this.getNewRailcar(),
 			candidateVerticals: [],
+			isSuggestionSelected: false,
+			searchValue: props.searchValue,
 		};
+		this.updateVerticalDataDebounced = props.bindDebounce( this.updateVerticalData );
 	}
 
+	componentDidUpdate( prevProps ) {
+		// The suggestion list should only be updated when the vertical suggestions results change.
+		// Note: it's intentional to use object reference comparison here.
+		// Since `verticals` props is connected from a redux state here, if the two references are identical,
+		// we can safely say that the two content are identical, thanks to the immutability invariant of redux.
+		if ( prevProps.verticals !== this.props.verticals ) {
+			// It's safe here to call setState() because we prevent the indefinite loop by the wrapping condition.
+			// See the official doc here: https://reactjs.org/docs/react-component.html#componentdidupdate
+			this.setSearchResults( this.props.verticals );
+		}
+	}
+
+	/**
+	 * Sets `state.results` with incoming vertical results, retaining previous non-user vertical search results
+	 * if the incoming vertical results contain only user-defined results.
+	 *
+	 * This function could better be performed in the backend eventually.
+	 *
+	 * @param {Array} results Incoming vertical results
+	 */
+	setSearchResults = results => {
+		if ( size( results ) ) {
+			const { candidateVerticals, isSuggestionSelected } = this.state;
+			// If the user is typing
+			// and the only result is a user input (non-vertical),
+			// and we have some previous results
+			// then concat that with the previous results and remove the last user input
+			if (
+				! isSuggestionSelected &&
+				! find( results, item => ! item.isUserInputVertical ) &&
+				1 < candidateVerticals.length
```

Comment:
FYI no need to use Yoda condition in calypso, but please feel free to keep it if it pleases you :)

Reply:
I know, it's a habit that's hard to break now, like ( spaces ) in all my ( parentheses ) 😆